src/imgui/ImGuiMachine.cc

```diff
@@ -142,6 +172,33 @@ void ImGuiMachine::paintSelectMachine(const MSXMotherBoard* motherBoard)
 			ImGui::Separator();
 		}
 
+		if (!recentMachines.empty()) {
+			ImGui::TextUnformatted("Recently used machines:"sv);
+			auto* info = findMachineInfo(recentMachines.front());
+			im::Combo("##recent", info->displayName.c_str(), [&]{
+				for (const auto& item : recentMachines) {
+					info = findMachineInfo(item);
+					bool ok = info && getTestResult(*info).empty();
```

Comment:
Here you test for `info != nullptr`, but above on line 177 you do not. I think we should either check both or check none.

This list is potentially populated with information from `imgui.ini`. It might be the case that between the time this was saved and when it's reloaded some machines have been deleted. So I think we should indeed check.

Ah, and when `info == nullptr` various things below can go wrong (line 183, 184, 185, 194).
* Line 183: `info->configName`, fallback to `item`?
* Line 184: `info->displayName`, also fallback to `item`?
* Line 184: in case of `!ok` we could pass `ImGuiSelectableFlags_Disabled` to `ImGui::Selectable()` (3rd parameter). Then lines 185-190 should never execute (and the test on `ok` inside is no longer needed). Just an idea I did not test this. We should also verify that disabled selectables can still be hovered (otherwise the user cannot see the error message)`
* Line 194: `*info`, maybe we should pass a pointer instead of a reference to `printConfigInfo()`, and then in case of a `nullptr` print a generic error message like `machine config doesn't exist`.

--

Or alternatively when loading `recentMachines` we could then check whether these machines actually exist, and if not then drop the item? This makes the situation the same as in the `Available machines` list box: then you can do `assert(info)`. (In theory it's still possible to delete a machine config file while openMSX is running. Such errors are only detected when you actually try to activate that machine).

Reply:
Should be fixed in the latest version.

After our discussion, I decided upon skipping items that are not known machines (no MachineInfo present) in the list that is displayed in the GUI.